vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/steps/TouchSteps.java

```diff
@@ -16,43 +16,76 @@
 
+import org.apache.commons.lang3.EnumUtils;
+import org.apache.commons.lang3.Validate;
 import org.jbehave.core.annotations.When;
+import org.jbehave.core.model.ExamplesTable;
 import org.openqa.selenium.WebElement;
 import org.vividus.bdd.mobileapp.model.SwipeDirection;
 import org.vividus.bdd.monitor.TakeScreenshotOnFailure;
 import org.vividus.bdd.steps.ComparisonRule;
 import org.vividus.bdd.steps.ui.validation.IBaseValidations;
 import org.vividus.mobileapp.action.KeyboardActions;
 import org.vividus.mobileapp.action.TouchActions;
+import org.vividus.selenium.IWebDriverProvider;
 import org.vividus.selenium.manager.GenericWebDriverManager;
 import org.vividus.ui.action.ISearchActions;
+import org.vividus.ui.action.JavascriptActions;
 import org.vividus.ui.action.search.Locator;
 
+import io.appium.java_client.android.nativekey.AndroidKey;
+import io.appium.java_client.android.nativekey.KeyEvent;
+import io.appium.java_client.android.nativekey.PressesKey;
+
 @TakeScreenshotOnFailure
 public class TouchSteps
 {
     private static final float VISIBILITY_TOP_INDENT_COEFFICIENT = 0.15f;
     private static final float VISIBILITY_BOTTOM_INDENT_COEFFICIENT = 0.25f;
 
+    private static final Map<String, String> ANDROID_KEYS = Map.ofEntries(
+        entry(" ", AndroidKey.SPACE.toString()),
+        entry("0", AndroidKey.DIGIT_0.toString()),
+        entry("1", AndroidKey.DIGIT_1.toString()),
+        entry("2", AndroidKey.DIGIT_2.toString()),
+        entry("3", AndroidKey.DIGIT_3.toString()),
+        entry("4", AndroidKey.DIGIT_4.toString()),
+        entry("5", AndroidKey.DIGIT_5.toString()),
+        entry("6", AndroidKey.DIGIT_6.toString()),
+        entry("7", AndroidKey.DIGIT_7.toString()),
+        entry("8", AndroidKey.DIGIT_8.toString()),
+        entry("9", AndroidKey.DIGIT_9.toString())
+        );
+
     private final TouchActions touchActions;
     private final KeyboardActions keyboardActions;
     private final IBaseValidations baseValidations;
     private final ISearchActions searchActions;
     private final GenericWebDriverManager genericWebDriverManager;
+    private final JavascriptActions javascriptActions;
+    private final IWebDriverProvider webDriverProvider;
 
     public TouchSteps(TouchActions touchActions, KeyboardActions keyboardActions, IBaseValidations baseValidations,
-            ISearchActions searchActions, GenericWebDriverManager genericWebDriverManager)
+            ISearchActions searchActions, GenericWebDriverManager genericWebDriverManager,
+            JavascriptActions javascriptActions, IWebDriverProvider webDriverProvider)
     {
         this.touchActions = touchActions;
         this.keyboardActions = keyboardActions;
         this.baseValidations = baseValidations;
         this.searchActions = searchActions;
         this.genericWebDriverManager = genericWebDriverManager;
+        this.javascriptActions = javascriptActions;
+        this.webDriverProvider = webDriverProvider;
```

Comment:
it looks like this class is overloaded, does it make sense to create a new class, e.g. `KeyboardSteps`?

Reply:
![image](https://user-images.githubusercontent.com/17816729/123795295-8af31500-d8ec-11eb-8ae4-5b6a98d77f6c.png)